server/handlers/contexts_handler.go

```diff
@@ -70,15 +95,27 @@ func (h *Handler) GetContext(w http.ResponseWriter, req *http.Request, _ *models
 	// 	return
 	// }
 
+	userID := uuid.FromStringOrNil(user.ID)
+	eventBuilder := events.NewEvent().FromUser(userID).FromSystem(*h.SystemID).WithCategory("connection").WithAction("retrieve")
+
 	token, ok := req.Context().Value(models.TokenCtxKey).(string)
 	if !ok {
 		http.Error(w, "failed to get token", http.StatusInternalServerError)
 		return
 	}
-
+	id := mux.Vars(req)["id"]
+	connectionID := uuid.FromStringOrNil(id)
+	eventBuilder.ActedUpon(connectionID)
 	h.log.Info("this is being used\n\n\n")
-	val, err := provider.GetK8sContext(token, mux.Vars(req)["id"])
+	val, err := provider.GetK8sContext(token, id)
 	if err != nil {
+		eventBuilder.WithSeverity(events.Error).WithDescription("Encountered an issue while retrieving Kubernetes context.").
+			WithMetadata(map[string]interface{}{
+				"error":                 err,
+				"probable_cause":        "You might have passed incorrect query params or there may be an issue with the connectivity between meshery and kubernetes.",
+				"suggested_remediation": "Please ensure that you've provided valid query params (page, pagesize, search, order), and Meshery can establish connectivity with kubernetes.",
```

Comment:
please change this to `suggested remedy`
```suggestion
				"suggested_remedy": "Ensure that you've provided valid query params (page, pagesize, search, order), and Meshery can establish connectivity with kubernetes.",
```

Reply:
is there not also another  kind of remedy message , our users may not understand the meaning of 
`provided valid query params (page, pagesize, search, order)`

Reply:
> is there not also another kind of remedy message , our users may not understand the meaning of `provided valid query params (page, pagesize, search, order)`

Yes, you're right user will be confused about query params :smile:

Reply:
`suggested_remediation` must remain the same.

The "Please" should be removed.